pypeit/core/wavecal/wvutils.py

```diff
@@ -199,81 +204,81 @@ def get_wave_grid(waves, masks=None, wave_method='linear', iref=0, wave_grid_min
     """
     c_kms = constants.c.to('km/s').value
 
-    if masks is None:
-        masks = waves > 1.0
-
-    if wave_grid_min is None:
-        wave_grid_min = waves[masks].min()
-    if wave_grid_max is None:
-        wave_grid_max = waves[masks].max()
-
-    dwave_data, dloglam_data, resln_guess, pix_per_sigma = get_sampling(waves)
-
-    # TODO: These tests of the string value should not use 'in', they should use ==
-    if ('velocity' in wave_method) or ('log10' in wave_method):
-        if dv is not None and dloglam is not None:
-            msgs.error('You can only specify dv or dloglam but not both')
-        elif dv is not None:
-            dloglam_pix = dv/c_kms/np.log(10.0)
-        elif dloglam is not None:
-            dloglam_pix = dloglam
-        else:
-            dloglam_pix = dloglam_data
-
-        # Generate wavelength array
-        wave_grid = wavegrid(wave_grid_min, wave_grid_max, dloglam_pix,
-                             spec_samp_fact=spec_samp_fact, log10=True)
-        loglam_grid_mid = np.log10(wave_grid) + dloglam_pix*spec_samp_fact/2.0
-        wave_grid_mid = np.power(10.0, loglam_grid_mid)
-        dsamp = dloglam_pix
+    if 'user_input' in wave_method:
+        wave_grid = wave_grid_input
+    else:
+        if masks is None:
+            masks = waves > 1.0
+
+        if wave_grid_min is None:
+            wave_grid_min = waves[masks].min()
+        if wave_grid_max is None:
+            wave_grid_max = waves[masks].max()
+
+        dwave_data, dloglam_data, resln_guess, pix_per_sigma = get_sampling(waves)
+
+        # TODO: These tests of the string value should not use 'in', they should use ==
+        if ('velocity' in wave_method) or ('log10' in wave_method):
```

Comment:
Don't do this to test string equality.  Instead use `==`.  I.e.:
```
if wave_method == 'velocity' or wave_method == 'log10':
```
but here, because there are two options, I would instead use:
```
if wave_method in ['velocity', 'log10']:
```

Reply:
Done.